refactoring/README.md

```diff
@@ -0,0 +1,15 @@
+# Refactoring
```

Comment:
I've read this file and it doesn't seem to add any valuable information.
We could move individual sentences related to individual arguments in the introductions of the relevant chapters if you think they are relevant and missing.

For example in the `idioms` introduction we can write something like:

```
We can use idioms to structure our code in a way that is understandable
```

if you think it would fit well.

Reply:
I think the person wanted to do more in these regards and I think it's quite useful, it's the content that will be added in the future that will make it more useful, for sure. Same for functional.

```
* `functional`
  
  * currying
  * higher-order functions
  * collections

* `refactoring`
  
  * refactoring from `unwrap` to `Result`
  * refactoring from `enum` to `Option`
  ```

Reply:
I've added this list to the Readme file as a TODO with refactoring and functional as new points

Reply:
So yeah, I agree with you that at this point it doesn't add too much valuable information, but I think it's nice to have as kind of a "placeholder" for the structure and to show people that there could be something added/PRed

Reply:
I prefer to have a clean version of the book on master.
If it's just a phrase like "TODO talk also about this" it's fine, but if it's an entire chapter that has to be written it's too much I think. For the reader it feels like the book is incomplete.
I prefer to open an issue to track those missing chapters, since those are no more that ideas or "future requests" at the moment.

So basically I would just totally remove this file from this PR, again, moving some phrases around if you think they might be valuable somewhere else.
What do you think?

Reply:
Hmm, I understand where you coming from and I partly agree for sure. On the other hand not every chapter in the repository needs to be rendered to HTML for the book. We can influence that with the SUMMARY.md file. So we could just not link these files in the book but have them in the repository already. Personally I would prefer this solution.

Reply:
I removed the link for refactoring-README.md from SUMMARY.md so it is marked as a draft in that regards (it will be greyed out in the book). Is that fine with you or would you rather completely remove the Headline from the TOC?

Reply:
![2021-01-02 05_58_46-Functional Programming - Rust Design Patterns – Mozilla Firefox-Screenshot](https://user-images.githubusercontent.com/14062932/103451073-9c51c080-4cbf-11eb-910f-e674021c895b.png)

Reply:
Like this it is now, which imho is fine. It's not a link.

![2021-01-02 06_01_18-Introduction - Rust Design Patterns – Mozilla Firefox-Screenshot](https://user-images.githubusercontent.com/14062932/103451097-0f5b3700-4cc0-11eb-9b4c-99cfd9b79d05.png)

Reply:
Honestly it looks really ugly :rofl:
If I were a reader, after I see this I would submit a PR to fix the link :sweat_smile: 
I mean..we don't even have a plan on what to put in this "Refactoring" section I think.
This file only contains a summary of the rest of the book :/
Let's open an issue to discuss *if* we want a refactoring chapter and what should be discussed there before adding a draft chapter in the book which will remain in this state for a long time I presume.

Reply:
Ok, i'll take it out of the summary then 🤣

Reply:
Should we commit this file in your opinion?

Reply:
I know there is not much in it already, but I really think it is a good idea to consider refactoring as a related part of software design (->patterns). 

"Refactoring is intended to improve the design, structure, and/or implementation of the software (its non-functional attributes), while preserving its functionality."

I think it would be good to start putting it as we should see the Book itself as a clean surface, the repository structure should be clean as well, but I assume we can fix that soonish and it will not stay long so empty.

Reply:
Also because there are `idoms` as well. I think the scope of this repository could really be around the process of good software design in Rust in the end.

Reply:
I feel like idioms are more related to design patterns with respect to refactoring.
But if you feel like this topic should be added it's ok :)